### PR TITLE
[ᚬrc/v0.15.z] fix: Ban peer when validate received block failed

### PR DESCRIPTION
**Pull request by TheWaWaR**

### CODE REVIEW

**Comment by TheWaWaR** [comment]
> I think we should not pass network context around, but pop up the error instead.

Agreed

File: sync/src/synchronizer/mod.rs [hunk 1]
```diff
@@ -167,15 +167,15 @@ impl<CS: ChainStore> Synchronizer<CS> {
...
-            return;
+            return Ok(());
         }
 
         match self.shared().get_block_status(&block.header().hash()) {
             BlockStatus::VALID_MASK => {
-                self.insert_new_block(peer, block);
+                self.insert_new_block(peer, block)?;
             }
             status => {
```
Comment (keroro520) [inline sync/src/synchronizer/mod.rs]: Why not return an error when `BlockStatus` is unexpected?

Comment (TheWaWaR) [reply sync/src/synchronizer/mod.rs]: I'm not sure it's a case to ban this peer

Comment (keroro520) [reply sync/src/synchronizer/mod.rs]: I am not sure, either. Keep as it is.

**Pull request merged**